src/Compilers/CSharp/Test/Semantic/Semantics/IteratorTests.cs

```diff
@@ -429,5 +429,79 @@ IEnumerable<int> I()
                 Diagnostic(ErrorCode.ERR_EmptyYield, "return").WithLocation(7, 15)
                 );
         }
+
+        [Fact]
+        [WorkItem(3825, "https://github.com/dotnet/roslyn/issues/3825")]
+        public void ObjectCreationExpressionSyntax_01()
+        {
+            var text = @"
+using System.Collections.Generic;
+
+class Test<TKey, TValue>
+{
+    public IEnumerator<KeyValuePair<TKey, TValue>> GetEnumerator(KeyValuePair<TKey, TValue> kvp)
+    {
+        yield return new KeyValuePair<TKey, TValue>(kvp.Key, kvp.Value);
+    }
+}";
+            var comp = CreateCompilationWithMscorlib45(text);
+            comp.VerifyDiagnostics();
+
+            var tree = comp.SyntaxTrees[0];
+            var node = (ObjectCreationExpressionSyntax)tree.GetRoot().DescendantNodes().Where(n => n is ObjectCreationExpressionSyntax).Single();
```

Comment:
Perhaps `.OfType<ObjectCreationExpressionSyntax>()` rather than `.Where(...)`. Then the cast can be removed.